webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx

```diff
@@ -54,34 +54,46 @@ const ExperimentActions: React.FC<Props> = ({ experiment, onClick, onSettled }:
 
   const handleArchive = useCallback((archive: boolean) => async (): Promise<void> => {
     setButtonStates(state => ({ ...state, archive }));
-    await archiveExperiment(experiment.id, archive);
-    onSettled();
-    setButtonStates(state => ({ ...state, archive: false }));
+    try {
+      await archiveExperiment(experiment.id, archive);
+      onSettled();
+    } finally {
```

Comment:
we don't catch potential exceptions here. this is an existing problem but we can catch it if we add a catch clause. feel free to leave for later